modules/tracing/mule-tracer-exporter-impl/src/main/java/org/mule/runtime/tracer/impl/exporter/OpenTelemetrySpanExporter.java

```diff
@@ -104,42 +232,186 @@ public void updateChildSpanExporter(SpanExporter childSpanExporter) {
-          && !openTelemetrySpan.getNoExportUntil().contains(getWithoutNamespace(childSpanExporter.getInternalSpan().getName()))) {
-        childOpenTelemetrySpanExporter.openTelemetrySpan = getNewOpenTelemetrySpan(internalSpan,
-                                                                                   new NoExportInitialSpanInfo(childOpenTelemetrySpanExporter
-                                                                                       .getOpenTelemetrySpan()
-                                                                                       .getNoExportUntil()),
-                                                                                   NO_EXPORTABLE_SERVICE);
+      if (!noExportUntil.isEmpty()
+          && !noExportUntil.contains(getNameWithoutNamespace(childSpanExporter.getInternalSpan().getName()))) {
+        childOpenTelemetrySpanExporter.parentSpanContext = parentSpanContext;
+        childOpenTelemetrySpanExporter.noExportUntil = noExportUntil;
+        childOpenTelemetrySpanExporter.spanContext = spanContext;
+        childOpenTelemetrySpanExporter.rootSpan = rootSpan;
+        childOpenTelemetrySpanExporter.exportable = false;
+      }
+
+      if (childOpenTelemetrySpanExporter.parentSpanContext == getInvalid()) {
+        childOpenTelemetrySpanExporter.parentSpanContext = spanContext;
       }
     }
+  }
 
+  @Override
+  public InternalSpan getInternalSpan() {
+    return internalSpan;
   }
 
-  private void setRootSpan(OpenTelemetrySpanExporter openTelemetrySpan) {
-    this.rootSpan = openTelemetrySpan;
+  @Override
+  public String getName() {
+    if (overriddenSpanName == null) {
+      return internalSpan.getName();
+    }
+
+    return overriddenSpanName;
   }
 
-  private String getWithoutNamespace(String name) {
-    int index = name.lastIndexOf(":");
-    if (index != -1) {
-      return name.substring(index + 1);
-    } else {
-      return name;
+  @Override
+  public SpanKind getKind() {
+    return spanKind;
+  }
+
+  @Override
+  public void onAdditionalAttribute(String key, String value) {
+    if (key.equals(SPAN_KIND)) {
+      spanKind = SpanKind.valueOf(value);
+    }
+
+    if (key.equals(STATUS)) {
+      StatusCode statusCode = StatusCode.valueOf(value);
+      statusData = StatusData.create(statusCode, null);
     }
   }
 
   @Override
-  public InternalSpan getInternalSpan() {
-    return internalSpan;
+  public SpanContext getSpanContext() {
+    return spanContext;
+  }
+
+  @Override
+  public SpanContext getParentSpanContext() {
+    return parentSpanContext;
+  }
+
+  @Override
+  public String getParentSpanId() {
+    return SpanData.super.getParentSpanId();
   }
 
+  @Override
+  public StatusData getStatus() {
+    return statusData;
+  }
+
+  @Override
+  public long getStartEpochNanos() {
+    return internalSpan.getDuration().getStart();
+  }
+
+  @Override
+  public Attributes getAttributes() {
+    return this;
```

Comment:
Why are you returning this?

Reply:
Because the OpentelemetrySpanExporter is working as the attributes holder (it implements Attributes from the optel sdk) so that no new instance is created and then the forEach method is invoked by optel.